### DISCUSSION TITLE
KeyVault permission model

### DISCUSSION THREAD

**Discussion by jonnyry** [microsoft/AzureTRE#3855] [Ideas]
Just wondering what people's thoughts are on using RBAC for controlling KeyVault permissions instead of Access Policies?

E.g. if you are frequently building and destroying a TRE instance, Access Policies need setting everytime for each KeyVault.

**Comment by marrobi** [comment]
Yes, would be great. Access Policies are old school.

**Comment by tamirkamara** [comment]
One thing to consider is that RBAC has latency challenges so I expect that will cause issues for deploying this solution.

**Reply by jonnyry** [reply]
Ah OK.. as in latency in applying an RBAC change?  https://learn.microsoft.com/en-us/answers/questions/524792/azure-rbac-propagation-latency

**Reply by tamirkamara** [reply]
Yes.
I know policies are considered legacy but will be with us for a long time. This solution can still try to make use of RBAC permissions but that will require some work and plenty of tests (although I might be wrong and it will be an easy change).
